modules/03_in_memory_storage/README.md

```diff
@@ -1,9 +1,482 @@
 # In memory storage
 
-TODO: write intro description
-TODO: write tutorial
-TODO: address secret key needing to remain secret <https://flask.palletsprojects.com/en/0.12.x/quickstart/#sessions>
-TODO: explain sessions
-TODO: explain app_version injection
+## Goal
 
-Potential additional topics to cover: testing and debugging
\ No newline at end of file
+Implement an interactive web app where you can:
+- upload words to the game dictionary;
+- play a game of guessing a word from the dictionary.
+
+## Tutorial
+
+> In this tutorial we will assume that you've completed the previous tutorial
+and already have a python project with `app.py`, `src/api.py` and `src/templates`.
+
+### Part 1. Uploading words
+
+#### Intro
+
+In the previous stage, we have implemented a so-called "static" website, where
+you can get pages from the server, but their content is always the same.
+
+Now it's time to **upload** data to the server! In our case, upload words to the dictionary.
+
+To do this, we'll need:
+1. a text field on the web page with a button "submit" (this will send data to the server)
+2. an endpoint on the server that would accept this data.
+
+#### 1. Create HTML page for word upload
+
+Let's create a web page that will contain a text field for the word to upload and a button `submit`.
+
+Create a new file `upload_word.html` in the folder `templates`:
+
+```shell
+touch templates/upload_word.html
```

Comment:
We covered creating files in earlier chapters and stopped giving direct instructions on this. Do you want to keep this in?

Reply:
True, removed!